hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/clientrm/RouterClientRMService.java

```diff
@@ -623,4 +650,22 @@ public void initUserPipelineMap(Configuration conf) {
         YarnConfiguration.DEFAULT_ROUTER_PIPELINE_CACHE_MAX_SIZE);
     this.userPipelineMap = Collections.synchronizedMap(new LRUCacheHashMap<>(maxCacheSize, true));
   }
+
+  private URL getRedirectURL() throws Exception {
+    Configuration conf = getConfig();
+    String webAppAddress = WebAppUtils.getWebAppBindURL(conf, YarnConfiguration.ROUTER_BIND_HOST,
+        WebAppUtils.getRouterWebAppURLWithoutScheme(conf));
+    String[] hostPort = StringUtils.split(webAppAddress, ':');
+    if (hostPort.length != 2) {
+      throw new YarnRuntimeException("Router can't get valid redirect proxy url");
+    }
+    String host;
+    if (null == hostPort[0] || hostPort[0].equals("") || hostPort[0].equals("0.0.0.0")) {
```

Comment:
Can we refactor this part of the code? This part of the code doesn't look easy to read.

Reply:
refactor `null == hostPort[0] || hostPort[0].equals("")`  to `StringUtils.isNotBlank(hostPort[0])`.
Is it ok?